app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.java

```diff
@@ -39,311 +37,437 @@
-
-  private final BehaviorSubject<CommentsData> commentsData = BehaviorSubject.create();
-  private final BehaviorSubject<String> currentCommentBody = BehaviorSubject.create();
-  private final BehaviorSubject<Void> dismissCommentDialog = BehaviorSubject.create();
-  private final BehaviorSubject<Boolean> enablePostButton = BehaviorSubject.create();
-  private final BehaviorSubject<Boolean> isFetchingComments = BehaviorSubject.create();
-  private final BehaviorSubject<Boolean> commentButtonHidden = BehaviorSubject.create();
-  private final BehaviorSubject<Pair<Project, Boolean>> showCommentDialog = BehaviorSubject.create();
-  private final PublishSubject<Void> showCommentPostedToast = PublishSubject.create();
-  private final PublishSubject<ErrorEnvelope> showPostCommentErrorToast = PublishSubject.create();
-
-  public final CommentsViewModelInputs inputs = this;
-  public final CommentsViewModelOutputs outputs = this;
-
-  @Override public void commentBodyChanged(final @NonNull String string) {
-    this.commentBodyChanged.onNext(string);
-  }
-  @Override public void commentButtonClicked() {
-    this.commentButtonClicked.onNext(null);
-  }
-  @Override public void commentDialogDismissed() {
-    this.commentDialogDismissed.onNext(null);
-  }
-  @Override public void loginSuccess() {
-    this.loginSuccess.onNext(null);
-  }
-  @Override public void nextPage() {
-    this.nextPage.onNext(null);
-  }
-  @Override public void postCommentClicked() {
-    this.postCommentClicked.onNext(null);
-  }
-  @Override public void refresh() {
-    this.refresh.onNext(null);
-  }
+      final Observable<User> currentUser = Observable.merge(
+              this.currentUser.observable(),
+              this.loginSuccess.flatMap(__ -> this.client.fetchCurrentUser().compose(neverError())).share()
+      );
 
-  @Override public @NonNull Observable<CommentsData> commentsData() {
-    return this.commentsData;
-  }
-  @Override public @NonNull Observable<String> currentCommentBody() {
-    return this.currentCommentBody;
-  }
-  @Override public @NonNull Observable<Void> dismissCommentDialog() {
-    return this.dismissCommentDialog;
-  }
-  @Override public @NonNull Observable<Boolean> enablePostButton() {
-    return this.enablePostButton;
-  }
-  @Override public @NonNull Observable<Boolean> isFetchingComments() {
-    return this.isFetchingComments;
-  }
-  @Override public @NonNull Observable<Boolean> commentButtonHidden() {
-    return this.commentButtonHidden;
-  }
-  @Override public @NonNull Observable<Pair<Project, Boolean>> showCommentDialog() {
-    return this.showCommentDialog;
-  }
-  @Override public @NonNull Observable<Void> showCommentPostedToast() {
-    return this.showCommentPostedToast;
-  }
-  @Override public @NonNull Observable<String> showPostCommentErrorToast() {
-    return this.showPostCommentErrorToast
-      .map(ErrorEnvelope::errorMessage);
+      final Observable<Either<Project, Update>> projectOrUpdate = intent()
```

Comment:
if you could run `make bootstrap` to grab our Kickstarter code style (or maybe you already have it! (Pref > Code Style > Scheme: set to Kickstarter IDE), then these tabs should all convert to our 2-space indentation style

Reply:
Great! I found the scheme in /script/style.